packager/media/formats/webm/buffer_mkv_writer.cc

```diff
@@ -0,0 +1,124 @@
+// copyright 2020 google inc. all rights reserved.
+//
+// use of this source code is governed by a bsd-style
+// license that can be found in the license file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+#include "packager/media/formats/webm/buffer_mkv_writer.h"
+
+namespace shaka {
+namespace media {
+
+BufferMkvWriter::BufferMkvWriter() : position_(0) {}
+
+BufferMkvWriter::~BufferMkvWriter() {}
+
+Status BufferMkvWriter::OpenBuffer() {
+  seekable_ = false;
+  position_ = 0;
+  return Status::OK;
+}
+
+Status BufferMkvWriter::OpenFile(const std::string& name) {
+  DCHECK(!file_);
+  file_.reset(File::Open(name.c_str(), "w"));
+  if (!file_)
+    return Status(error::FILE_FAILURE, "Unable to open file for writing.");
+
+  // This may produce an error message; however there isn't a seekable method
+  // on File.
+  seekable_ = file_->Seek(0);
+  position_ = 0;
+  return Status::OK;
+}
+
+bool BufferMkvWriter::WriteToFile(const std::string& file_name) {
+
+  if (file_) {
+    LOG(ERROR) << "File " << file_->file_name() << " is open.";
+    return false;
+  }
+  file_.reset(File::Open(file_name.c_str(), "w"));
+  if (!file_) {
+    LOG(ERROR) << "Failed to open file " << file_name;
+    return false;
+  }
+  seekable_ = file_->Seek(0);
+  return segment_buffer_.WriteToFile(file_.get()).ok();
+}
+
+Status BufferMkvWriter::CloseFile() {
+  const std::string file_name = file_->file_name();
+  if (!file_.release()->Close()) {
+    return Status(
+        error::FILE_FAILURE,
+        "Cannot close file " + file_name +
+            ", possibly file permission issue or running out of disk space.");
+  }
+  return Status::OK;
+}
```

Comment:
I think we can combine the three functions to one function: `WriteToFile(const std::string& file_name)`.